eslint/index.js

```diff
@@ -0,0 +1,57 @@
+module.exports = {
+  rules: {
+    'proptype-definition': {
+      create(context) {
+        return {
+          AssignmentExpression(node) {
+            // helper func which targets possible react components
+            const findDeclaration = singleNode => {
+              if (
+                singleNode.type.match(/[Function|Variable|Class|Export(Named|Default)]Declaration/)
+              ) {
+                // deal with default or named export
+                if (singleNode.declaration) {
+                  const { declarations } = singleNode.declaration;
+                  if (
+                    (declarations && declarations[0].id.name === propertyTypeNode.name) ||
+                    singleNode.declaration.id.name === propertyTypeNode.name
+                  ) {
+                    return true;
+                  }
+                  // deal with variable declaration or function statement
+                } else if (
+                  (singleNode.id && singleNode.id.name === propertyTypeNode.name) ||
+                  (singleNode.declarations &&
+                    singleNode.declarations[0].id.name === propertyTypeNode.name)
+                ) {
+                  return true;
+                }
+              }
+              return false;
```

Comment:
This is a good use-case for `lodash.get`!

```js
if (!singleNode.type.match(/[Function|Variable|Class|Export(Named|Default)]Declaration/)) {
  return false;
}

const isDefaultOrNamedExport =
  get(singleNode, 'declaration.id.name') === propertyTypeNode.name ||
  get(singleNode, 'declaration.id.name.declarations[0].id.name') ===
    propertyTypeNode.name;

if (isDefaultOrNamedExport) return true;

const isVariableOrFunctionStatement =
  get(singleNode, 'id.name') === propertyTypeNode.name ||
  get(singleNode, 'declarations[0].id.name') === propertyTypeNode.name;

if (isVariableOrFunctionStatement) return true;
```

Reply:
okay, I will make the change later today